drivers/usb/udc/Kconfig.ambiq

```diff
@@ -0,0 +1,34 @@
+ # Copyright (c) 2024 Ambiq Micro Inc.
```

Comment:
What controller is it? Are there other vendors using the same controller design?

Reply:
Hi @jfischer-no, the handling in this `udc_ambiq.c` is specific to Ambiq as we need to access the USB controller functionalities via Ambiq's USB HAL. I believe that the logic here is not reusable for USB controller in other vendor's chip.